### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by zarva6596**
[DEMO LINK](https://zarva6596.github.io/react_clock/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,24 +2,24 @@ import React from 'react';
...
+    date: new Date().toLocaleTimeString(),
+  };
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+  render() {
+    setTimeout(() => {
+      this.setState({ date: new Date().toLocaleTimeString() });
+    }, 1000);
```
Comment (Totowa95) [inline src/App.js:13]: - start the timer only when the component is added to the page
нужно использовать componentDidMount()

**Review by Totowa95 — CHANGES_REQUESTED** [review]

**Review by Totowa95 — APPROVED** [review]

File: src/App.js [hunk 2]
```diff
@@ -2,24 +2,28 @@ import React from 'react';
...
-};
+  render() {
+    return (
+      <div className="box">
+        <div className="box__time">
+          {this.state.date}
+        </div>
+      </div>
+    );
+  }
```
Comment (vkryvytskyy) [inline src/App.js:26]: better to create separate component <Clock /> and move all logic there

File: src/App.js [hunk 3]
```diff
@@ -2,24 +2,28 @@ import React from 'react';
...
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  render() {
+    return (
+      <div className="box">
+        <div className="box__time">
+          {this.state.date}
```
Comment (vkryvytskyy) [inline src/App.js:22]: use destructuring

File: src/App.js [hunk 4]
```diff
@@ -2,24 +2,28 @@ import React from 'react';
...
+    date: new Date().toLocaleTimeString(),
+  };
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+  componentDidMount() {
+    setInterval(() => {
+      this.setState({
+        date: new Date().toLocaleTimeString(),
```
Comment (vkryvytskyy) [inline src/App.js:13]:
```suggestion
        date: new Date(),
```

File: src/App.js (hunk 3 above)
Comment (vkryvytskyy) [inline src/App.js:22]:
```suggestion
          {date.toLocaleTimeString()}
```
this way you need to use this method only once

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 2 above)
Comment (zarva6596) [reply src/App.js]: done

**Review by vkryvytskyy — APPROVED** [review]

**Pull request closed without merging**